src/core/src/preprocess/preprocess_impls.cpp

```diff
@@ -193,14 +194,6 @@ bool InputInfo::InputInfoImpl::build(const std::shared_ptr<Model>& model,
     if (node.get_partial_shape() != context.model_shape()) {
         need_validate = true;  // Trigger revalidation if input parameter shape is changed
     }
-    // Check final shape
-    OPENVINO_ASSERT(node.get_partial_shape().compatible(context.model_shape()),
```

Comment:
Can we disable this check only for gray transformation?

Reply:
It was a mistake from my side, this check is fully right for RGB->Gray conversion too